File: 150-逆波兰表达式求值.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        mystack = []
        a = ('+', '-', '*', '/')
        n = len(tokens)
        i = 0

        while i < n:
            s = tokens[i]
            if s in a:
                num1 = int(mystack.pop())
                num2 = int(mystack.pop())
                if s == '+':
                    num = num2 + num1
                elif s == '-':
                    num = num2 - num1
                elif s == '*':
                    num = num2 * num1
                else:
                    num = int(num2 / num1)
                mystack.append(num)
            else:
                mystack.append(int(s)) 
            i += 1
        
        return mystack.pop()

File: 150-逆波兰表达式求值.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        int i = 0;
        int n = tokens.size();
        int num, num1, num2;
        vector<string> a = {"+", "-", "*", "/"};
        string s;
        stack <int> mystack;

        while(i < n){  
            s = tokens[i];
            if(find(a.begin(), a.end(), s) != a.end()){
                num1 = mystack.top();
                mystack.pop();
                num2 = mystack.top();
                mystack.pop();
                if(s == "+")
                    num = num2 + num1;
                else if(s == "-")
                    num = num2 - num1;
                else if(s == "*")
                    num = num2 * num1;
                else
                    num = num2 / num1;
                mystack.push(num);
            }

            else
                mystack.push(stoi(s));
            
            i += 1;
        }

        return mystack.top();
    }
};
